### PR TITLE
Support for multi dataset in metadata panel

### PR DESCRIPTION
**Pull request by ptbrowne**
* Showing information for all the cubes in the metadata panel
* Grouping the dimensions by cube in the metadata panel

### CODE REVIEW

File: app/components/metadata-panel.tsx [hunk 1]
```diff
@@ -406,17 +412,47 @@ const Header = ({ onClose }: { onClose: () => void }) => {
...
+  const cubesMetadata = data?.dataCubesMetadata;
+  if (!cubesMetadata) {
+    return null;
+  }
   return (
     <TabPanel className={classes.tabPanel} value="general">
-      <DataSetMetadata dataSetIri={datasetIri} dataSource={dataSource} />
+      {fetching ? <Loading /> : null}
+      {error ? (
+        <Error>{`${error instanceof Error ? error.message : error}`}</Error>
```
Comment (bprusinowski) [inline app/components/metadata-panel.tsx:447]: So `error` could be something different than `Error`? 🤯

File: app/components/dataset-metadata.tsx [hunk 2]
```diff
@@ -19,7 +19,7 @@ import { Icon } from "@/icons";
 import { useLocale } from "@/locales/use-locale";
 import { makeOpenDataLink } from "@/utils/opendata";
 
-export const DataSetMetadata = ({
+export const DatasetMetadata = ({
```
Comment (bprusinowski) [inline app/components/dataset-metadata.tsx:22]: 👍

File: app/components/metadata-panel.tsx [hunk 3]
```diff
@@ -470,6 +478,37 @@ const TabPanelData = ({ dimensions }: { dimensions: Component[] }) => {
     return dimensions.map((d) => ({ label: d.label, value: d }));
   }, [dimensions]);
 
+  const locale = useLocale();
+  const grouped = groupBy(
+    dimensions.flatMap((d): Component[] => {
+      if ("originalIris" in d) {
+        return (
+          d.originalIris?.map((o) => ({
```
Comment (bprusinowski) [inline app/components/metadata-panel.tsx:486]: Nit: do we need `?` here, when `originalIris` property exists?

**Review by bprusinowski — APPROVED** [review]
LGTM! 👌 

Two things noticed when reviewing deployment preview:
- it looks that dimension values are not shown in the metadata panel (I guess it happened when I introduced `loadValues` property some time ago, it also doesn't work on TEST 😨). Let me know if I should take a look at this – maybe it would make sense to only fetch them when user actually goes to the dimension preview, to not fire potentially expensive queries upfront?
- `joinBy` dimension is not translated 👀

File: app/components/metadata-panel.tsx (hunk 3 above)
Comment (ptbrowne) [reply app/components/metadata-panel.tsx:486]: The property exists but could be undefined ;)

Comment (ptbrowne) [reply app/components/metadata-panel.tsx]: I moved the check to the if ;)

File: app/components/metadata-panel.tsx (hunk 1 above)
Comment (ptbrowne) [reply app/components/metadata-panel.tsx:447]: Yes, every thing can be throw ;)

```
> const hello = () => { try { throw 'world' } catch (e) { console.log('got thrown', e) } }
undefined
> hello()
got thrown world
```

File: app/components/metadata-panel.tsx [hunk 4]
```diff
@@ -418,7 +422,33 @@ const TabPanelData = ({
...
+        d
+      ): {
+        label: string;
+        value: Component;
+        isJoinByDimension: boolean;
+      }[] => {
+        if (
+          "isJoinByDimension" in d &&
+          d.isJoinByDimension &&
+          "originalIris" in d
```
Comment (bprusinowski) [inline app/components/metadata-panel.tsx:437]: Nit: I think we have this check in several places now, maybe would make sense to extract to something like `isJoinByDimension`?

File: app/locales/de/messages.po [hunk 5]
```diff
@@ -1445,32 +1449,32 @@ msgid "time-units."
 msgstr ""
 
 #: app/configurator/components/ui-helpers.ts
-msgstr "time-units.Day"
-msgstr "Tag"
+msgid "time-units.Day"
+msgstr ""
```
Comment (bprusinowski) [inline app/locales/de/messages.po:1453]: Should this be removed? 🤔

**Review by bprusinowski — APPROVED** [review]
LGTM 🎉 

Just one small thing from deployment preview: it looks like the joined dimension label is missing translation?
Screenshot 2024-03-12 at 15 12 29

File: app/charts/combo/combo-line-dual-state.tsx [hunk 6]
```diff
@@ -53,6 +53,8 @@ export type ComboLineDualState = CommonChartState &
     maxRightTickWidth: number;
   };
 
+const mean = (arr: number[]) => arr.reduce((a, b) => a + b, 0) / arr.length;
```
[the remark this answers is not in the document]
Comment (ptbrowne) [reply app/charts/combo/combo-line-dual-state.tsx]: Done after ;)

File: app/components/metadata-panel.tsx (hunk 4 above)
Comment (ptbrowne) [reply app/components/metadata-panel.tsx]: Thanks, good idea, I extracted the guard in [`3e01901` (#1399)](https://github.com/visualize-admin/visualization-tool/pull/1399/commits/3e01901b169690c5e25a83fb324cd4089908b15c)

File: app/locales/de/messages.po (hunk 5 above)
Comment (ptbrowne) [reply app/locales/de/messages.po:1454]: Thanks, removed.

**Comment by ptbrowne** [comment]
Thanks @bprusinowski , I've added the missing translations.

**Pull request merged**